src/main/webapp/app/components/tabs/curation-data-import-tab/CurationDataImportTab.tsx

```diff
@@ -0,0 +1,602 @@
+    authStore,
+    firebaseGeneService,
+    firebaseMetaService,
+    alterationStore,
+    isGermline,
+    data,
+    mutationList: Mutation[],
+    vusList: VusObjList,
+  ) {
+    return {
+      data: data.data,
+      ...(await saveMutation(
+        authStore,
+        firebaseGeneService,
+        firebaseMetaService,
+        alterationStore,
+        isGermline,
+        data as DataRow<GermlineMutationDI>,
+        mutationList,
+        vusList,
+      )),
+    };
+  },
+  async [DataImportType.GERMLINE_MUTATION](
+    authStore,
+    firebaseGeneService,
+    firebaseMetaService,
+    annotateAlterations,
+    isGermline,
+    data,
+    mutationList: Mutation[],
+    vusList: VusObjList,
+  ) {
+    return {
+      data: data.data,
+      ...(await saveMutation(
+        authStore,
+        firebaseGeneService,
+        firebaseMetaService,
+        annotateAlterations,
+        isGermline,
+        data as DataRow<GermlineMutationDI>,
+        mutationList,
+        vusList,
+      )),
+    };
+  },
+};
+
+const CurationDataImportTab = observer(
+  ({ authStore, firebaseGeneService, firebaseGeneReviewService, firebaseMetaService, alterationStore }: ICurationToolsTabProps) => {
+    const [isGermline, setIsGermline] = useState(false);
+    const [selectedDataTypeOption, setSelectedDateTypeOption] = useState<DataImportTypeSelectOption | null>(null);
+    const [fileHeaders, setFileHeaders] = useState<string[]>([]);
+    const [fileRows, setFileRows] = useState<DataRow<DataImportObj>[]>([]);
+    const [fileUploadErrorMsg, setFileUploadErrorMsg] = useState('');
+    const [importStatus, setImportStatus] = useState<'' | 'uploaded' | 'importing' | 'imported'>('');
+    const fileUploadInputRef = useRef<HTMLInputElement>(null);
+    const IMPORT_STATUS_HEADER_KEY = 'status';
+    const IMPORT_STATUS_ERR_MSG_HEADER_KEY = 'message';
+    const ROW_COLUMN_KEY = 'row_id';
+
+    // Initiate a file reader and attach the onload event after uploading the file from the user
+    const fileReader = new FileReader();
+    fileReader.onload = event => {
+      setFileUploadErrorMsg('');
+      const text = (event.target?.result ?? '') as string;
+      tsvFileToArray(text);
+    };
+
+    const requiredCols: RequiredKeys<DataImportObj>[] =
+      selectedDataTypeOption !== null ? getRequiredColumns(selectedDataTypeOption.value) : [];
+    const optionalCols: OptionalKeys<DataImportObj>[] =
+      selectedDataTypeOption !== null ? getOptionalColumns(selectedDataTypeOption.value) : [];
+
+    // reset import status after changing gene type and data type
+    useEffect(() => {
+      resetFileStatus();
+    }, [isGermline, selectedDataTypeOption]);
+
+    const resetFileStatus = () => {
+      setFileUploadErrorMsg('');
+      setImportStatus('');
+      setFileHeaders([]);
+      setFileRows([]);
+
+      if (fileUploadInputRef.current) {
+        fileUploadInputRef.current.value = null as unknown as string;
```

Comment:
I remember you should be able to set to `""` to clear the selected file. Then you can get rid of the type assertion.